Sources/Plugins/AttachmentManager/AttachmentManager.swift

```diff
@@ -121,6 +121,17 @@ open class AttachmentManager: NSObject, InputPlugin {
         })
     }
     
+    /// Performs an animated removal of all attachments
+    open func removeAllAttachments() {
+        let attachments = self.attachments
+        self.attachments.removeAll()
+        self.reloadData()
+        for i in 0..<attachments.count {
```

Comment:
Can we change the behavior a little please? This feels strange...

```
for index in 0..<self.attachments.count {
    let attachment = attachments[index]
    self.attachments.remove(at: index)
    self.delegate?.attachmentManager(self, didRemove: attachment, at: index)
}
```
The main reason is that `attachments[i]` might be nil after you call `removeAll`.